src/Columns/IColumn.h

```diff
@@ -397,6 +397,16 @@ class IColumn : public COW<IColumn>
     /// It affects performance only (not correctness).
     virtual void reserve(size_t /*n*/) {}
 
+    /// Requests the removal of unused capacity.
+    /// It is a non-binding request to reduce the capacity of the underlying container to its size.
+    virtual void shrinkToFit()
+    {
+      /// Create a new column with the same values and move it into the initial column
+      auto temporary_copy = cloneEmpty();
+      temporary_copy->insertRangeFrom(*this, 0, size());
```

Comment:
This looks way too inefficient. Check if it's worth adding specializations in various column types.

Reply:
Optimized